### PR TITLE
[DNM] scripts: runner: nrfjprog: don't do full erase by default

### LINKED ISSUE #6147: "ninja flash" cannot be used with DFU-capable applications
Hi,

as a user, when I do

```
$ cd bootloader_build_dir
$ ninja flash

$ cd app_build_dir
$ ninja flash
```

The second flash command will erase the bootloader written during the first flash command.

This makes "ninja flash" useless for applications that use a bootloader.

"ninja flash" is a useful feature that should be preserved for DFU-able applications.

### PR DESCRIPTION
**Pull request by mbolivar**
By default, use a sector erase to flash boards with nrfjprog.py.

To allow getting the old behavior, add an --erase flag that works the
exact same way as the corresponding jlink.py argument: if present, a
full flash erase is done before programming. If absent, only a sector
erase is done.

Fixes #6147

### CODE REVIEW

**Review by carlescufi — COMMENTED** [review]
I am fine with this in principle, but I have never used the `sectorerase` flag in the 6+ years I've used nrfjprog. I will ask for others to comment on this just in case, in particular @nvlsianpu and @cvinayak.

**Comment by mbolivar** [comment]
> I am fine with this in principle, but I have never used the sectorerase flag in the 6+ years I've used nrfjprog. I will ask for others to comment on this just in case.

FWIW, I tested this with and without --erase on nrf52840-pdk, and it worked.

**Comment by carlescufi** [comment]
>FWIW, I tested this with and without --erase on nrf52840-pdk, and it worked.

Yep, I'm worried about side-effects related to `UICR` or other registers. But I don't know enough about them so I will leave it to the experts.

**Comment by mbolivar** [comment]
> Yep, I'm worried about side-effects related to UICR or other registers. But I don't know enough about them so I will leave it to the experts.

Makes sense. For reference, I based this on a comment from @SebastianBoe here: https://github.com/zephyrproject-rtos/zephyr/issues/6147#issuecomment-365281591

**Review by SebastianBoe — APPROVED** [review]
This is desired behaviour, great stuff.

"If --sectorerase is given, only the targeted non-volatile memory pages excluding UICR will be erased." --nrfjprog --help

**Comment by carlescufi** [comment]
@mbolivar I have tweaked it a bit to erase the UICR for nRF52x

**Review by nvlsianpu — APPROVED** [review]

**Comment by carlescufi** [comment]
recheck

**Comment by mbolivar** [comment]
> @mbolivar I have tweaked it a bit to erase the UICR for nRF52x

No objections from me -- we OK to merge?

**Comment by mbolivar** [comment]
Marking this do not merge (DNM) per discussions on IRC

**Comment by mbolivar** [comment]
I've ported this to west: https://github.com/zephyrproject-rtos/west/pull/2

**Pull request closed without merging**
